### ISSUE TITLE
The YOLO test in `utils.py`

### ISSUE THREAD

**Issue by matthewkirby** [LSSTDESC/CLMM#193]
Sometimes, tests on travis crash. Never locally, only ever on travis. The test that crashes is this one from `test_utils.py` in the function `test_compute_radial_averages()`

```python
# Test a much larger, random sample with unevenly spaced bins
binvals = 10.0 * np.random.random(100)
xbins2 = [0.0, 10.0 * np.random.random(), 10.0]
inbin1 = binvals[(binvals > xbins2[0]) & (binvals < xbins2[1])]
inbin2 = binvals[(binvals > xbins2[1]) & (binvals < xbins2[2])]
assert_allclose(compute_radial_averages(binvals, binvals, xbins2, error_model='std/n'),
                [[np.mean(inbin1), np.mean(inbin2)], [np.mean(inbin1), np.mean(inbin2)],
                 [np.std(inbin1)/len(inbin1), np.std(inbin2)/len(inbin2)]], **TOLERANCE)
assert_allclose(compute_radial_averages(binvals, binvals, xbins2, error_model='std'),
                [[np.mean(inbin1), np.mean(inbin2)], [np.mean(inbin1), np.mean(inbin2)],
                 [np.std(inbin1), np.std(inbin2)]], **TOLERANCE)
```

The output that travis sees is
```
x and y nan location mismatch:
x: array([[4.834679,      nan],
               [4.834679,      nan],
               [2.904419, 0.      ]])
y: array([[4.834679,      nan],
               [4.834679,      nan],
               [2.904419,      nan]])
```

None of these elements should be nan so something is going wrong. I am going to take a look at this (so we can fix the sometimes crashes).

My intent with this test was to 
1. Generate 100 random numbers between 0 and 10
2. Generate one random number between 0 and 10 and then make bin edges [0, random, 10]. This will give us two bins with a random width.

My expectation is that the nans are coming when the bins are something like [0,9.8,10] and then there are no random numbers drawn between [9.8,10]. This seems unlikely as it keeps coming up. I may have a typo in my intents as well. Maybe a fix is to increase the number of objects in binvals, less likely to have zero objects in a bin? Or we could write a big list of random numbers (100, or 1000 or something and put the file in `tests/data/filename.txt`), read that in, say the bins are like [0.0, 6.3, 10.] or something and just manually compute it and test this rather than doing it randomly.

**Comment by matthewkirby** [comment]
I just ran pytest 20 times locally on master. It worked every time. The crash rate seems to be at least 50% (probably higher) on travis.

**Comment by matthewkirby** [comment]
Okay, travis uses parallelization to run tests. The crashes are because this test is not thread-safe.

**Comment by matthewkirby** [comment]
Resolved in #194
